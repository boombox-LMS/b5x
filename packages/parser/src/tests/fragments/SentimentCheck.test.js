const {
  exampleSentimentCheckMarkup,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/SentimentCheck");
const { RawFragmentSchema } = require("@b5x/types");
const { z } = require("zod");

describe("The SentimentCheck fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleSentimentCheckMarkup
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed sentiment-check fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed sentiment-check fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed sentiment-check fragment uses the expected data schema", () => {
    const validator = () => {
      z.object({}).parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
