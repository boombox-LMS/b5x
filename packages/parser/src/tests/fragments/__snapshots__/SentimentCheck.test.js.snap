// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SentimentCheck fragment builds as expected The parsed sentiment-check fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
  ],
  "contentType": "SentimentCheck",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "baa130f7632fbad6ef8c78f06b77402c",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
